File: search/linear_search/javascript/linear_search.js
function linearSearch(array, value) {
  return array.indexOf(value);
}

// test
[
  [ [1, 2, 3, 4], 2, 1],
  [ [1, 2, 3, 4], 4, 3],
  [ [1, 2, 3, 4], 5, -1],
  [ [1, 3, 4, 5, 6], 2, -1 ]
].forEach(test => {
  console.assert(linearSearch(test[0], test[1]) == test[2]);
});

File: search/linear_search/scala/linear_search.scala
/**
*	Recursive linear search implemented in scala
*	unnecessarily complex
*/	
def linear_search(lst:Array[Int],idx:Int,key:Int):Int={
	if(lst.length == 0 ) return -1 else
		if(lst(idx) == key) return idx else
			return linear_search(lst,idx+1,key) 
}

def test(){

	val l = Array(12,3,4,5,6,7)

	assert(linear_search(l,0,4) == 2)
}